tests/integration/test_charm.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import pathlib
+import secrets
+
+from ops.model import ActiveStatus, Application
+import pytest
+
+from pytest_operator.plugin import OpsTest
+import jenkins
+
+from .types import JenkinsCredentials
+
+PLUGINS_DIR = pathlib.Path("/var/lib/jenkins/plugins")
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_active(app: Application):
+    """
+    arrange: given charm has been deployed and is idle
+    act: when the status is checked
+    assert: then the status is active and a message which contains running has been set
+    """
+    assert app.units[0].workload_status == ActiveStatus.name
+    assert "running" in app.units[0].workload_status_message
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_debian_stable_in_sources(app: Application):
+    """
+    arrange: given charm has been deployed and is idle
+    act: when /etc/apt/sources.list is checked
+    assert: then it contains debian-stable
+    """
+    souces_list_contents = await app.units[0].ssh("cat /etc/apt/sources.list")
+
+    assert "debian-stable" in souces_list_contents, "LTS not in sources.list"
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_tools_installed(app: Application):
+    """
+    arrange: given charm has been deployed and is idle
+    act: when checking for the python3-testtools package
+    assert: then it is found
+    """
+    dpkg_output = await app.units[0].ssh("dpkg -l python3-testtools")
+
+    assert "ii" in dpkg_output, "No tool installation found"
```

Comment:
ii?

Reply:
`ii`` appears in the `dpkg -l <package name>` output only if the package `<package name>` is installed. I have added a comment to clarify this